### ISSUE TITLE
After Cloth Fill the Editor Crashes

### ISSUE THREAD

**Issue by schmidt-florian** [FlaxEngine/FlaxEngine#1844]
**Issue description:**
When you fill the Cloth Flag in the Example Project and press play the Editor Crashes.

**Steps to reproduce:**
1. Open Physics Samples
2. Press the fill command on the Flag
3. Press Play

https://github.com/FlaxEngine/FlaxEngine/assets/14238126/9985b1ec-391f-454d-a027-bee317e6ca27

Log Entry is unfortionally clean.
The Sample Project is freshly cloned from Github.

**Flax version:**
1.7

**Comment by MineBill** [comment]
The error happens here:
```cpp
void Cloth::Rebuild()
{
#if WITH_CLOTH
    // Remove old
    if (IsDuringPlay())
        PhysicsBackend::RemoveCloth(GetPhysicsScene()->GetPhysicsScene(), _cloth);
    DestroyCloth();

    // Create new
    CreateCloth();
    if (IsDuringPlay() && _cloth)
===>    PhysicsBackend::AddCloth(GetPhysicsScene()->GetPhysicsScene(), _cloth); <===
#endif
}
```

After filling the entire flag, the cloth gets rebuilt. However, after DestroyCloth destroys the cloth and sets _cloth to nullptr, CreateCloth does some extra checks to see if the entire cloth is "fixed" and returns early to prevent unnecessary cloth simulation. As such, _cloth stays null, which causes an exception at the marked line above.

Attempting to remove the relevant block of code at:

https://github.com/FlaxEngine/FlaxEngine/blob/536be6c6cf0dda775680adf2a4f62d66ca5b8e42/Source/Engine/Physics/Actors/Cloth.cpp#L531-L538C6 

will result in a weird crash when loading the level:
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Repeat 2 times:
--------------------------------
   at FlaxEngine.Level.<Internal_LoadSceneFromBytes>g____PInvoke|204_0(Byte*, Int32*)
--------------------------------
   at FlaxEngine.Level.Internal_LoadSceneFromBytes(Byte[], Int32 ByRef)
   at FlaxEngine.Level.LoadSceneFromBytes(Byte[])
   at FlaxEditor.Utilities.DuplicateScenes.CreateScenes()
   at FlaxEditor.States.PlayingState.OnEnter()
   at FlaxEngine.Utilities.StateMachine.SwitchState(FlaxEngine.Utilities.State)
   at FlaxEditor.States.EditorStateMachine.SwitchState(FlaxEngine.Utilities.State)
   at FlaxEngine.Utilities.StateMachine.GoToState(FlaxEngine.Utilities.State)
   at FlaxEngine.Utilities.StateMachine.GoToState[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]()
   at FlaxEditor.Modules.SimulationModule.OnUpdate()
   at FlaxEditor.Editor.Update()
   at FlaxEngine.Interop.NativeInterop+Invoker+InvokerNoRet0`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MarshalAndInvoke(System.Object, FlaxEngine.Interop.ManagedHandle, IntPtr)
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(FlaxEngine.Interop.ManagedHandle, FlaxEngine.Interop.ManagedHandle, IntPtr, IntPtr)
```

The easiest solution is to add a null check before adding the cloth to the actor but it gets iffy with tooling because you wont be able to paint the cloth in the editor(there are null checks for the cloth, you have to press Reset and the select the cloth again to start painting).

**Comment by mafiesto4** [comment]
Fixed in 1a254afd4f610e80dfc8e21ec158d9a726263e35. I added some minor usability improvements to new Cloth in c0a8d29453d1388c5f71c7814e9a3d5af5d742a9
